### PR TITLE
Stats Widget: Introduce new Stats Widget with a basic chart

### PR DESCRIPTION
**Pull request by dognose24**
Related to #75102 

## Proposed Changes

* Add `widget/index.jsx` as the Stats Widget application root.
* Add `widget/mini-chart.jsx`
* Update existing chart-related components to fit the design.

The `widget-loader` is about `~9KB`, and the main load is `~310KB` atm. The `wpcom` client takes up a huge place (50KB+) there, which will be replaced with a simple wrapper in a follow up PR.

截圖 2023-04-01 上午4 25 43

## Testing Instructions

* Open your local Jetpack site.
* Check out the branch of the Jetpack PR: https://github.com/Automattic/jetpack/pull/29929/.
* Build Jetpack if necessary: `jetpack build plugins/jetpack`.
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` under `calypso/apps/odyssey-stats`.
* Navigate to `/wp-admin/index.php?odyssey_widget=1`.
* Ensure the Odyssey Widget displays in line with the design.
* Ensure i18n works
* Ensure the chart respect your site timezone
* Ensure periods in Calypso Stats still works okay
* Click the bars
* Ensure it takes you to correct Odyssey Stats with correct params

image

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by kangzj** [comment]
The chart is working now, but I'll see if we could omit some of the redux stuff, so that we could make the bundle slimmer. 🪓

File: apps/odyssey-stats/src/app.scss [hunk 1]
```diff
@@ -1,3 +1,85 @@
 @import "./styles/theme.scss";
 @import "./styles/wp-admin.scss";
 @import "./styles/typography.scss";
+
```
Comment (kangzj) [inline apps/odyssey-stats/src/app.scss:4]: These styles are moved from `styles/wp-admin.scss`

**Comment by a8ck3n** [comment]
This is working in my testing but I'm seeing some CSS being impacted on the dashboard.

SCR-20230405-rcdt

1. Font size in the Screen Options panel has been increased from 13px to 16px.
2. Left-side border is no longer correct.
3. Widget titles have changed. Appear the same size but no longer bolded.
4. List item styling has changed. Normally these are rendered with a "dot" marker.

File: apps/odyssey-stats/src/widget/mini-chart.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+const CHART_VISITORS = {
+	attr: 'visitors',
+	label: translate( 'Visitors', { context: 'noun' } ),
+};
+
+const CHARTS = [ CHART_VIEWS, CHART_VISITORS ];
+
+const MiniChart = ( { siteId, quantity = 7, gmtOffset, odysseyStatsBaseUrl } ) => {
```
Comment (jsnmoon) [inline apps/odyssey-stats/src/widget/mini-chart.jsx:26]: The hover tooltip is missing the month and day formatting:

image

File: apps/odyssey-stats/src/widget/index.scss [hunk 3]
```diff
@@ -1,7 +1,3 @@
 @import "../styles/theme.scss";
```
Comment (jsnmoon) [inline apps/odyssey-stats/src/widget/index.scss:1]: I see the same problems reported by @a8ck3n; we'll want to limit our styles to our widget and avoid affecting the rest of the dashboard.

**Review by jsnmoon — APPROVED** [review]
This is a good initial prototype. I'd like us to fix the two issues I've noted (style leak and hover tooltip formatting), but it'd be fine to handle this in a follow-up PR.

File: apps/odyssey-stats/src/widget/mini-chart.jsx (hunk 2 above)
Comment (kangzj) [reply apps/odyssey-stats/src/widget/mini-chart.jsx:26]: Good catch! This is related to https://github.com/Automattic/wp-calypso/pull/75255 and I'll address in a follow up PR. 👍

File: apps/odyssey-stats/src/widget/index.scss (hunk 3 above)
Comment (kangzj) [reply apps/odyssey-stats/src/widget/index.scss:1]: Created [an issue](https://github.com/Automattic/wp-calypso/issues/75360) to track it and we'll address in a follow up PR.

**Pull request merged**